### PR TITLE
Upgrade the SymfonyBundleTest

### PR DESCRIPTION
**Pull request by bocharsky-bw**
The `nyholm/symfony-bundle-test` prevents to upgrade to Symfony 6 completely as `symfony/framework-bundle` because it's installed as 5.4 for the current version SymfonyBundleTest version, see installed deps here: https://github.com/php-translation/symfony-bundle/runs/8168054961?check_suite_focus=true

I tried to upgrade it to the v2.0 and fixed broken tests so that this bundle could use all Symfony 6 packages finally.

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
If anybody has an idea how to fix those remaining 2 failed tests: https://github.com/php-translation/symfony-bundle/runs/8169007599?check_suite_focus=true - please, share.

**Comment by robbedg** [comment]
It seems like the session needs to be set manually:

```
        $this->testKernel->boot();
        $request = Request::create('/foobar');

        // Create and set session
        $session = new Session(new MockFileSessionStorage());
        $request->setSession($session);
        $this->testKernel->getContainer()->get(Activator::class)->setSession($session);

        // Activate the feature
        $this->testKernel->getContainer()->get(Activator::class)->activate();

        $response = $this->testKernel->handle($request);
```

**Comment by bocharsky-bw** [comment]
I was able to fix the remaining tests here - I think this PR is ready for the final review!

**Review by welcoMattic — APPROVED** [review]
:tada:

File: phpunit.xml.dist [hunk 1]
```diff
@@ -12,7 +12,8 @@
         >
     <php>
         <env name="ENV" value="test" />
-        <env name="SYMFONY_DEPRECATIONS_HELPER" value="verbose=1" />
+        <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[direct]=29&amp;verbose=1" />
```
Comment (bocharsky-bw) [inline phpunit.xml.dist:15]: We still have some deprecations that would be good to fix in separate PRs, I suppress them for now with this line to have GA actions completely green

**Review by rvanlaak — APPROVED** [review]

File: Tests/Functional/BundleInitializationTest.php [hunk 2]
```diff
@@ -21,8 +21,10 @@ class BundleInitializationTest extends BaseTestCase
 {
     public function testRegisterBundle(): void
     {
-        $this->bootKernel();
-        $container = $this->getContainer();
+        $kernel = $this->testKernel;
+        $kernel->boot();
+        $container = $kernel->getContainer();
```
[the remark this answers is not in the document]
Comment (bocharsky-bw) [reply Tests/Functional/BundleInitializationTest.php:26]: Yeah, but the problem is that I somehow cannot made this work with the parent `$kernel` so I created a custom `testKernel` property in `BaseTestCase` and therefore we have to call `getContainer()` on it.

If anyone could rewrite it with the parent `$kernel` and `self::getContainer()` and get tests passed - it would be cool, I agree... but I'm really giving up with it
